src/main/java/com/faforever/moderatorclient/api/domain/MapService.java

```diff
@@ -93,17 +93,30 @@ public MapService(FafApiCommunicationService fafApi) {
 
         ladder1v1Maps.forEach(ladder1v1Map -> ladder1v1Map.getMapVersion().setLadder1v1Map(ladder1v1Map));
 
-        List<Map> result = ladder1v1Maps.stream()
+        List<MapVersion> ladder1v1MapVersions = ladder1v1Maps.stream()
                 .map(Ladder1v1Map::getMapVersion)
-                .map(MapVersion::getMap)
                 .collect(Collectors.toList());
-        log.trace("found {} maps", result.size());
+        List<Map> preResult = ladder1v1MapVersions.stream()
+                .map(mapVersion -> {
+                    Map map = mapVersion.getMap();
+                    map.setVersions(map.getVersions().stream().filter(ladder1v1MapVersions::contains).collect(Collectors.toList()));
+                    return map;
+                })
+                .collect(Collectors.toList());
+        ArrayList<Map> result = new ArrayList<>(preResult);
+        for (Map map : preResult) {
+            //erase duplicates
+            if (result.stream().filter(map1 -> map.getId().equals(map1.getId())).count() > 1) {
+                result.remove(map);
+            }
+        }
+        log.trace("found {} maps", preResult.size());
```

Comment:
1. The Set
As you set I use the id instead of equals because equal method won't work here, I check that. We 
might wanna change that as well.
2. Returning Map instead of MapVersion
I changed it because of problems that were there. You add one map version, the map was added with all its versions and version that were not included were shown empty. And if you added two versions of the same map the map was shown twice. So I corrected all those fallacies. I only show versions that are in the ladder map pool and I don't show map duplicates any longer. From the way the tree view is structured it needs Map objects and also I did not change the return type I only corrected the data output of the method.

Reply:
I don't think I should change anything here should I?

Reply:
Oh that was due to some Lombok annotation leftovers. Fixed in https://github.com/FAForever/faf-java-commons/commit/6623f1f59c8bcb0d7c33423708cbf169e7c65638